ihp-sg13g2: Minor fixes to enabled BLOCKS

dnltz opened the pull request:
- Update `sg13g2.map` file and use latest one from PDK
- Fix lib handling in ihp-sg13g2 config.mk
- Update I2C GPIO Expander and add a (dummy) macro

dnltz committed:
flow: platforms: ihp-sg13g2: Update lib file handling

Call the variable ADDITIONAL_TYP_LIBS instead of ADDITIONAL_LIBS
and assign correct values to it. Also provide TYP_LIB_FILES next
to LIB_FILES with the same content.

Signed-off-by: Daniel Schultz <dnltz@aesc-silicon.de>

dnltz committed:
flow: platforms: ihp-sg13g2: Update sg13g2.map

Update sg13g2.map file with latest one from IHP's Open PDK.

Signed-off-by: Daniel Schultz <dnltz@aesc-silicon.de>

oharboe commented on flow/scripts/generate_abstract.tcl:
```diff
@@ -21,7 +21,15 @@ if {$design_stage >= 4} {
+  }
+  unset corner
+} else {
+  log_cmd write_timing_model $::env(RESULTS_DIR)/$::env(DESIGN_NAME)_typ.lib
```
is the change in .lib file intentional? It will break all users, no?

dnltz replied:
Yes, that's for purpose because no corner should mean typical. I updated the Makefile to handle that change. This script is only called when creating blocks.

oharboe replied:
This is a breaking change for users of ORFS, such as bazel-orfs. The output name is part of the ORFS interface/contract.

oharboe replied:
Regarding changing the external API/contract of generate_abstract.

Reading the example in OpenSTA, I do see the _typ.lib naming convention:

![image](https://github.com/user-attachments/assets/9cd03a98-58f3-4805-a145-cdfe873b1280)

oharboe reviewed (changes requested):
split out ORFS generic changes to a separate PR

dnltz committed:
flow: designs: ihp-sg13g2: Update I2C GPIO Expander

Move the I2cDeviceCtrl into a macro to have one chip design
with an embedded macro generated by OR.

Signed-off-by: Daniel Schultz <dnltz@aesc-silicon.de>
